### DISCUSSION TITLE
How to put data from DB, to show without using tables.

### DISCUSSION THREAD

**Discussion by EpsilonCommunity** [Budibase/budibase#4486] [Help] [answered]
Hi, wondering if there's a way to have data shown as Paragraphs/Headings etc.

So that a page can be dynamical populated, without using a table.

**Comment by mjashanks** [comment, marked as answer]
So - yes, if i'm understanding you correctly, you can

- Enter static text into paragraphs and headings
or
- Use a non-table datasource, like a rest API, to populate the elements.

You say "dynamically populated" - but dynamic based on what? You need some sort of data there to be "dynamic" with, don't you?

**Reply by EpsilonCommunity** [reply]
Database Table of customers for example with a /:id page of each customer.
I don't want a table of their information.

**Reply by mjashanks** [reply]
Oh you mean you want to display data, without using the `Table` component, in  the design section?

You should use a
```
> Data Provider
    > Repeater
        > other components
```

The Data provider will pull data from your datasource. The repeater will display components for every row returned by the data provider.

If you only want to "Repeat" on one row, then you need to 
1. add an appropriate filter to your data provider.... e.g. `_id equals {{ URL.id }}`
2. on the data provider, set the `limit` property to `1`

**Reply by EpsilonCommunity** [reply]
Thanks all sorted.
